### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by Nejjari001**
## In this milestone of the Math Magician App, we implemented the following:
- Set up React Testing Library.
-  Wrote unit tests for the files operate.js and calculate.js using Jest.
-  Created unit tests for all React components.
-  Used Jest snapshots to test the components.
-  Used React Testing Library to simulate user interaction.
- Ran all the full tests suite using Jest and ensured that all tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "@babel/core": "^7.20.2",
```
Comment (HAMMAS-SALEEM) [inline package.json:41]: readme

**[Optional]**

- [x] Your test from App.test.js file is not passing 😨 Kindly fix that 👍

File: src/logic/__test__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import calculate from '../calculate';
+
+describe('testing the calculate function', () => {
+  const obj = {
+    total: 0,
+    next: null,
+    operation: null,
+  };
+
+  // Addition
```
Comment (HAMMAS-SALEEM) [inline src/logic/__test__/calculate.test.js:18]: test

- [ ] You are missing these requirements shown in the image above ⬆️ You are not testing your components with snapshots 😨 Kindly do that 👍

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Nejjari001, @demix007 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @Nejjari001 and @demix007,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by demix007** [comment]
### Hello @julie-ify 👋🏼 

Thank you for taking out time to review and approve this PR.

File: src/logic/__test__/calculate.test.js (hunk 2 above)
Comment (demix007) [reply src/logic/__test__/calculate.test.js:18]: ## Hello @HAMMAS-SALEEM  👋🏼 
 Thanks for taking out time to review this PR. The required and optional changes have been duly noted and necessary changes will be implemented.

**Comment by Nejjari001** [comment]
## Thanks @julie-ify 💯 👍 for your kind and fast review

**Comment by demix007** [comment]
## Hello @Nejjari001 😎 

A big kudos to you for the immense effort and contribution you put into the completion of this project. You are amazing 💯

**Pull request merged**